### DISCUSSION TITLE
compiled model inside a package

### DISCUSSION THREAD

**Discussion by FelicienLL** [metrumresearchgroup/mrgsolve#975] [General]
Hello, 
I hope everything is fine, 

Is there an _easy_ way to store an already compiled (and ready to simulate) model object inside a package ? Since it is not possible to build a package with executable files, I guess we need some workarounds... I tried to modify the `house()` function to work with a model of my own but it fails. The point is that I'm really not familiar enough with the underlying concepts (compiling, shared objects, dll, etc...) to know if I'm just missing something simple, or if it is far too complicated to implement... 

I'm asking because I'd like to implement a compiled model in mapbayr in order to ease and speed unit testing, as well as using it for examples, as you do with `house()` in mrgsolve. 

Thanks in advance for your help

Félicien

**Comment by kylebaron** [comment]
Hi @FelicienLL  - 

It is super complicated. A long time ago I had this same idea of pre-compiling a bunch of models in a package and it would be there and ready to go.  It still might be possible but let's think about it. 

So one thing that you could do today is build the model and cache the result.  This means that in a given R session, you should only have to build the model once. Also for your internal testing, you could keep a bunch of models in cache in a stable location and refer to that cache if you're testing locally and build a new one if the build or tests are happening on CRAN. 

Use `mread_cache()` and use the `soloc` argument; or set `options(mrgsolve.soloc = ...)` ; you could set up some code that would look for an environment variable pointing to the stable cache (local) and if that wasn't set, build in `tempdir()`; this could be used every time you build / load a model so it could work for every example, every test etc. 

Kyle

**Comment by FelicienLL** [comment]
Ok, thank you for your answer, I'll drop this idea and have a closer look to the cache option. Thanks again
Félicien
